WebDevStudios/wd_s
# Hotfix/mobile js issues

### DESCRIPTION ###
I've noticed that we've had to fix this, or versions of this, on projects. This resolves an issue where the triggers for click/touchstart events on mobile are triggering when they shouldn't.

For instance:
1. If I tap on my screen on an element with a click event attached to it in an attempt to swipe up and down the page, the click event is triggered
2. If I have a modal open and attempt to close the modal, if the spot I tap on the screen happens to fall on the modal trigger I am stuck in an endless loop of the modal closing and reopening

This seems to be resolved by removing `touchstart` from our events all together and just relying on `click`. They both may have been necessary at one point, but they no longer seem to be. `click` seems to do the job as expected on mobile while also avoiding the unexpected triggers from taking place.

This should be tested extensively on mobile devices, of course! I've tested it on my end and we've implemented this previously with no issue, but there's obviously no harm in _more_ testing.

A part of this fix also adds `return false;` to the end of events to take another step toward stopping the multiple event trigger issue.

This also adds a fix to the modal as it pertains to the `esc` key. Previous to this update, hitting the `esc` key anywhere on the page was putting the focus on the modal trigger. This update resolves that, so if you hit escape elsewhere on the page for whatever reason, you won't be jumped to the modal trigger.

### SCREENSHOTS ###
https://dl.dropbox.com/s/px084bvgl56h73m/wd_s-mobile-js.gif?dl=0

(link if the GIF is too large to embed: `https://www.dropbox.com/s/px084bvgl56h73m/wd_s-mobile-js.gif?dl=0`)

### OTHER ###

### STEPS TO VERIFY ###
Checkout the branch and try triggering actions on your mobile devices. I narrowed it down to these elements to test:
- Hamburger/Mobile Menu
- Header Button
- Accordion
- Modal

Also ensure these continue to work as expected on desktop.

On desktop, also try hitting the `esc` key to escape the Header Search, or just hit the `esc` key whenever. Confirm that hitting the `esc` key does _not_ always place the focus on the modal trigger.

Compare to `master` to see the differences.
